src/client.c

```diff
@@ -485,9 +485,7 @@ bool set_pool(PgSocket *client, const char *dbname, const char *username, const
 			 * see if the global auth_user is set and use that.
 			 */
 			if (!client->db->auth_user_credentials && cf_auth_user) {
-				client->db->auth_user_credentials = find_global_credentials(cf_auth_user);
-				if (!client->db->auth_user_credentials)
-					client->db->auth_user_credentials = add_global_credentials(cf_auth_user, "");
+				client->db->auth_user_credentials = find_exist_or_add_new_global_credentials(cf_auth_user, "");
```

Comment:
This should fail too right if it's NULL?

Reply:
Should I write something like this

```c
slog_error(client, "set_pool(): failed to allocate a new global credentials");
disconnect_client(client, true, "bouncer resources exhaustion");
return false;
```
?

I will rename these new functoin and will return to this problem.